pkg/detectors/meraki/meraki.go

```diff
@@ -0,0 +1,115 @@
+package meraki
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	apiKey = regexp.MustCompile(detectors.PrefixRegex([]string{"meraki"}) + `([0-9a-f]{40})`)
+)
+
+func (s Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+
+	return defaultClient
+}
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"meraki"}
+}
+
+func (s Scanner) Description() string {
+	return "Cisco Meraki is a cloud-managed IT solution that provides networking, security, and device management through an easy-to-use interface." +
+		"Meraki APIs make it possible to rapidly deploy and manage networks at scale, build on a platform of intelligent, cloud-connected IT products, and engage with users in powerful new ways."
+}
+
+// FromData will find and optionally verify Meraki API Key secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	// uniqueMatches will hold unique match values and ensure we only process unique matches found in the data string
+	var uniqueMatches = make(map[string]bool)
+
+	for _, match := range apiKey.FindAllStringSubmatch(dataStr, -1) {
+		if len(match) != 2 {
+			continue
+		}
+
+		uniqueMatches[strings.TrimSpace(match[1])] = true
```

Comment:
This is a strange practice established in earlier detector code. At best, it is completely pointless; at worst, it silently breaks the code if the pattern changes.
https://github.com/trufflesecurity/trufflehog/pull/2264#discussion_r1443382434

TrimSpace also isn't required unless the capture group includes whitespace.
```suggestion
	uniqueMatches[match[1]] = true
```

Reply:
TrimSpace can be removed. About the pattern, are you referring to index `1` we use from match?

Reply:
> About the pattern, are you referring to index 1 we use from match?

The pattern and `match[1]` are fine. I'm referring to the `if len(match) != 2` check.

Reply:
Okay that is now removed 😎